### PR TITLE
Occigen15

### PR DESCRIPTION
**Pull request by aminiussi**
Some changes to build with the intel compiler (15) on occigen (RH &.3), based on lib g++ 4.8.5.

### CODE REVIEW

File: test/api/range/segmented_input_range.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ STF_CASE_TPL("iteration with prologue", STF_NUMERIC_TYPES)
   for(std::size_t i=0;i<ref.size();i++) ref[i] = T(1);
 
   auto data = boost::make_iterator_range(ref.data()+1, ref.data()+ref.size());
-  auto pr = segmented_input_range(data);
+  auto pr = segmented_input_range<pack<T>::static_size>(data);
```
Comment (jfalcou) [inline test/api/range/segmented_input_range.cpp]: this is rather verbose. If think this is related to #426 we should have an explicit yet more palatable way to fix this.
My idea is to have `segmented_input_range(p)`, `s_i_range(p, int_<Card{})` and maybe `s_i_range(p, sse_ )` to mean either use the default cardinal, this exact cardinal OR the cardinal prefered by SSE architecture.
the `int_<C>` is also nice as it's compatible with the `1_c` syntax

What do you think of this ?

The rest of the PR is fine or me though.

Comment (aminiussi) [reply test/api/range/segmented_input_range.cpp]: Or a more functional flavor, like `s_i_range(card<N>(p)), s_i_range(sse_card(p))`, meaning _p with cardinality..._. It also separate the range vs iterator  overload from the cardinality overload, which could be easier as/if more overloads comes down the road..
Of course, keeping the s_i_range(p) version.
But I don't know what would be more "natural" for the intensive user.

Comment (jfalcou) [reply test/api/range/segmented_input_range.cpp]: I like this idea in fact, I'll give it a deep thought

**Comment by jfalcou** [comment]
Would you mind :  
 - remove the local fix for segmented range
 - rebase

adn we'll take care of getting a quick resolution to #426 instead

**Comment by aminiussi** [comment]
I'll do that tomorrow in the morning.

**Comment by aminiussi** [comment]
Ok, so as far as the hidden var goes, it was a mistake in my remote management:
I clone on a new machine at cines, usually all my branches trac origin (me) except for develop which trac numscale. As a result my develop lost trac and I forgot to set the numscale remote on that cluster...
Forget it, the hidden stuf can be closed.

**Pull request closed without merging**